src/object_store/src/object/opendal_engine/opendal_s3.rs

```diff
@@ -60,4 +59,38 @@ impl OpendalObjectStore {
             engine_type: EngineType::S3,
         })
     }
+
+    /// Creates a minio client. The server should be like `minio://key:secret@address:port/bucket`.
+    pub fn with_minio(server: &str) -> ObjectResult<Self> {
+        let server = server.strip_prefix("minio://").unwrap();
+        let (access_key_id, rest) = server.split_once(':').unwrap();
+        let (secret_access_key, mut rest) = rest.split_once('@').unwrap();
+        let endpoint_prefix = if let Some(rest_stripped) = rest.strip_prefix("https://") {
+            rest = rest_stripped;
+            "https://"
+        } else if let Some(rest_stripped) = rest.strip_prefix("http://") {
+            rest = rest_stripped;
+            "http://"
+        } else {
+            "http://"
+        };
+        let (address, bucket) = rest.split_once('/').unwrap();
+        let mut builder = S3::default();
+        builder
+            .bucket(bucket)
+            .region("custom")
+            .access_key_id(access_key_id)
+            .secret_access_key(secret_access_key)
+            .endpoint(&format!("{}{}", endpoint_prefix, address));
+
+        builder.disable_config_load();
+        let op: Operator = Operator::new(builder)?
+            .layer(LoggingLayer::default())
+            .layer(RetryLayer::default())
+            .finish();
```

Comment:
It is better to keep the same [configs ](https://github.com/risingwavelabs/risingwave/blob/f554c62dc5477fb3e4501468dd567c8d3861b2a2/src/object_store/src/object/opendal_engine/opendal_s3.rs#L47) for the regular s3 engine and minio. With aws sdk, we use the [same configs](https://github.com/risingwavelabs/risingwave/blob/f554c62dc5477fb3e4501468dd567c8d3861b2a2/src/object_store/src/object/s3.rs#L538) to initialize the http client for both regular s3 and minio. Can we extract the common logics out to make the behavior consistent?